src/main/java/com/project/trav/service/TicketService.java

```diff
@@ -17,52 +20,69 @@
 @Service
 @RequiredArgsConstructor
 public class TicketService {
-    private final TicketMapper ticketMapper;
-    private final UserRepository userRepository;
-    private final TicketRepository ticketRepository;
-    private static final String NOT_FOUND_ERROR ="Ticket was not found by id";
 
-    public List<TicketDto> getTickets(){return ticketMapper.toTicketDtos(ticketRepository.findAll());}
+  private final TicketMapper ticketMapper;
+  private final UserRepository userRepository;
+  private final RaceRepository raceRepository;
+  private final TicketRepository ticketRepository;
+  private static final String NOT_FOUND_ERROR = "Ticket was not found by id";
 
-    public TicketDto getTicket(Long id) {
-        return ticketMapper.toTicketDto(findByIdTicket(id));
-    }
-    public void addTicket(TicketDto ticketDto){ticketRepository.save(ticketMapper.toTicket(ticketDto));}
+  public List<TicketDto> getTickets() {
+    return ticketMapper.toTicketDtos(ticketRepository.findAll());
+  }
 
-    public void deleteTicket(Long id){
-        existByIdTicket(id);
-        ticketRepository.deleteById(id);
-    }
-    public void updateTicket(TicketDto ticketDto, Long id){
-        existByIdTicket(id);
-        ticketRepository.save(ticketMapper.toTicket(ticketDto));
-    }
-    public void buyTicket(Long ticketId, String username){
-        prepareTicket(TicketStatus.BOUGHT,ticketId,username);
-    }
-    public void bookTicket(Long ticketId,String username){
-        prepareTicket(TicketStatus.BOOKED,ticketId,username);
-    }
-    private Ticket findByIdTicket(Long id) {
-        return ticketRepository.findById(id).orElseThrow(()->{
-            throw new EntityNotFoundByIdException(NOT_FOUND_ERROR);
-        });
+  public TicketDto getTicket(Long id) {
+    return ticketMapper.toTicketDto(findByIdTicket(id));
+  }
+  public void addTicket(TicketDto ticketDto) {
+    validPlace(ticketDto);
+    ticketRepository.save(ticketMapper.toTicket(ticketDto));
+  }
 
+  public void deleteTicket(Long id) {
+    findByIdTicket(id);
+    ticketRepository.deleteById(id);
+  }
+
+  public void updateTicket(TicketDto ticketDto, Long id) {
```

Comment:
Could you please check:
Update existing ticket with place==null and placeClass ==null